src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -614,6 +616,12 @@ public void start(Job project, boolean newInstance) {
 
         // Create a new timer task if there is a URL
         if (dynamicTriggerConfiguration) {
+            // We *must* update list of projects immediately *before* we can be labeled as "started".
+            // When using imperative Jenkinsfiles, every run will "update" the job's configuration.
+            // When the job's configuration is updated, all triggers are stopped and restarted.
+            // Thus, if we do *not* update the project list, then Gerrit events in quick succession
+            // will be missed due to the start/stop cycle.
+            updateTriggerConfigURL();
```

Comment:
This will severely increase Jenkins startup and job save times.
We need to find a different approach at solving this problem.
For example adding a flag that `dynamicGerritProjects` is being updated and have `isInteresting` wait for that flag to clear.

Reply:
I don't see why it's a problem to have a job's trigger start only when it's ready to start.  I expect the trigger to only be ready when it has loaded the dynamic configuration; I don't think that anyone looking to use this plugin would think that it makes sense for the trigger to be basically useless and unpredictable with Jenkinsfile imperative pipelines.

I suppose that we could have a global cache that maps the URL to its data (or that maps the job to its data) so that the information would persist across trigger restarts.

If you don't want to do that, the only other quick solution that I can think of would be to block within `isInteresting` until the dynamic project list has been loaded (also with a static global lock unless you can figure out how to make Jenkins actually instantiate the local one for `isInteresting`).  I don't know what kind of effects that would have (that is, if `isInteresting` takes seconds to run instead of practically instantly).

Reply:
It's Jenkins startup time that I'm most concerned with. Potentially thousands of jobs wanting to contact external services and retrieve data, that's why it was put in a separate thread in the first place.

Reply:
> I suppose that we could have a global cache that maps the URL to its data (or that maps the job to its data) so that the information would persist across trigger restarts.

I have this functionality already in my repo. It works perfectly, let me fill PR.
Also look like task that polls dynamic configuration is never stopped, but trigger update fills new task.

Reply:
Having `isInteresting` take a bit of time to run from time to time is ok, it is designed for that so if throughput gets slow you can increase the number of threads serving incoming events. Though in this case it would only perform tiny bit better since in the end all would wait for that one block on the same jobs waiting for the dynamic projects to load. But when that's done it should continue to flow.

Reply:
@tekkamanendless are you OK with using/working around the cache in #353 ? If I should merge that first for you to use?

Reply:
This is now done in a separate thread from the `start` thread, and anything that needs to access the dynamic list of projects waits until the list is ready.